core/src/test/java/com/linecorp/armeria/server/ReconfigureTlsTest.java

```diff
@@ -0,0 +1,101 @@
+
+import java.security.cert.CertificateException;
+import java.security.cert.X509Certificate;
+import java.time.Instant;
+import java.time.temporal.ChronoUnit;
+import java.util.Date;
+import java.util.concurrent.atomic.AtomicReference;
+
+import org.junit.jupiter.api.Test;
+
+import com.linecorp.armeria.client.BlockingWebClient;
+import com.linecorp.armeria.client.ClientFactory;
+import com.linecorp.armeria.client.WebClient;
+import com.linecorp.armeria.common.AggregatedHttpResponse;
+import com.linecorp.armeria.common.HttpResponse;
+import com.linecorp.armeria.common.HttpStatus;
+import com.linecorp.armeria.internal.common.util.SelfSignedCertificate;
+
+class ReconfigureTlsTest {
+
+    @Test
+    void shouldUpdateTlsSettings() throws CertificateException {
+        final SelfSignedCertificate oldCert =
+                new SelfSignedCertificate(Date.from(Instant.parse("2022-01-01T00:00:00.00Z")),
+                                          Date.from(Instant.now().plus(10, ChronoUnit.DAYS)));
+        final AtomicReference<X509Certificate> sslContextRef = new AtomicReference<>();
+        final Server server =
+                Server.builder()
+                      .https(0)
+                      .tls(oldCert.certificate(), oldCert.privateKey())
+                      .service("/", (ctx, req) -> {
+                          sslContextRef.set((X509Certificate) ctx.sslSession().getLocalCertificates()[0]);
+                          return HttpResponse.of("OK");
+                      })
+                      .build();
+
+        server.start().join();
+
+        final BlockingWebClient client0 =
+                WebClient.builder("https://127.0.0.1:" + server.activeLocalPort())
+                         .factory(ClientFactory.builder()
+                                               .tlsCustomizer(sslContextBuilder -> {
+                                                   sslContextBuilder.trustManager(
+                                                           oldCert.certificate());
+                                               })
+                                               .build())
+                         .build()
+                         .blocking();
+        final AggregatedHttpResponse response = client0.get("/");
+        assertThat(response.status()).isEqualTo(HttpStatus.OK);
+        assertThat(sslContextRef.get().getNotBefore()).isEqualTo(oldCert.cert().getNotBefore());
+
+        final SelfSignedCertificate newCert =
+                new SelfSignedCertificate(Date.from(Instant.parse("2023-01-01T00:00:00.00Z")),
+                                          Date.from(Instant.now().plus(10, ChronoUnit.DAYS)));
+        server.reconfigure(sb -> {
+            sb.tls(newCert.certificate(), newCert.privateKey())
+              .service("/", (ctx, req) -> {
+                  sslContextRef.set((X509Certificate) ctx.sslSession().getLocalCertificates()[0]);
+                  return HttpResponse.of("OK");
+              });
```

Comment:
optional, I think it would be nice to assert that an exception is raised if reconfigure doesn't configure an ssl context

i.e. original port supports https, but a `sslContext` isn't configured so an IAE is raised.
```
            sb
              .service("/", (ctx, req) -> {
                  sslContextRef.set((X509Certificate) ctx.sslSession().getLocalCertificates()[0]);
                  return HttpResponse.of("OK");
              });
```

Reply:
Shouldn't we warn about such a case? 🤔

Reply:
Oops, sorry about it. Please never mind the comment. 😓